core branch descriptions (#76152)
ansible/ansible: docs/docsite/rst/dev_guide/core_branches_and_tags.rst

> .. _core_branches_and_tags:

Reviewer: If you choose NOT to move all of this into the development_process.rst, then please add `:orphan:` to the top of this page to prevent docs build warnings.

******************************************

> ``ansible-core`` Project Branches and Tags

Reviewer: ```suggestion
``ansible-core`` project branches and tags
```

Reviewer: We're still inconsistent in the docs, but we are supposed to use 'Sentence case' for headings, meaning only the first word is capitalized (though not in this instance since it's the package name).

******************************************

> ``devel`` Branch

Reviewer: ```suggestion
``devel`` branch
```

================
All new development on the next version of ``ansible-core`` occurs exclusively in the ``devel`` branch,
and all bugfixes to prior releases must first be merged to devel before being backported to one or more stable branches
for inclusion in servicing releases. Around the Beta 1 milestone, a new ``stable-X.Y`` branch is cut from ``devel``,
which is then updated to host development of the ``X.Y+1`` release. External automated testing of Ansible content from
``devel`` is not generally recommended.

> ``stable-X.Y`` Branches

Reviewer: ```suggestion
``stable-X.Y`` branches
```

=======================
All ``ansible-core`` ``X.Y.Z`` releases are created from a corresponding ``stable-X.Y`` branch. A
release's stable branch is typically cut from ``devel`` around ``X.Y.0 beta 1`` (when the release is feature complete).
All further bugfixes (no enhancements!) must be made against ``devel`` and backported to applicable stable branches.

> ``vX.Y.Z`` Tags

Reviewer: ```suggestion
``vX.Y.Z`` tags
```

===============
Each ``ansible-core vX.Y.Z`` release is tagged from the release commit in the corresponding ``stable-X.Y`` branch,
allowing access to the exact source used to create the release. As of ``ansible-core`` 2.13, the auto-generated GitHub

> tarball of the tag contents is the considered the official canonical release artifact.

Reviewer 2: Would you expect downstream distribution packaging to rely on GitHub instead of PyPI for the source tarball, then ?

Reviewer 3: I suppose that depends on the definition of downstream, but I see no problems with that.  We're still figuring things out, but there will come a time where an upstream gets shorter support than "downstream" (Red Hat) and we will no longer publish releases to PyPI, however downstream RH builds will continue, using GH for the source.

Reviewer: more of a side comment  - when we do get to that point (RH keeps going but pypi stops) - we'll need to consider what we do for docs. Porting guide specifically comes to mind, and the changelogs - if the pypi stopped at say X.Y.5, but RH keeps going for another handful of dot releases, do we keep updating the porting guide/changelogs? do we stop? Do we put some kind of special marker to say X.Y.6 release and beyond are only available through RH subscription?  Just things to consider when the time comes.

> tarball of the tag contents is the considered the official canonical release artifact.

Reviewer: ```suggestion
tarball of the tag contents is considered the official canonical release artifact.
```

.. _milestone_branch:

> ``milestone`` Branch

Reviewer: ```suggestion
``milestone`` branch
```

====================
A slow-moving stream of the ``devel`` branch, intended for external testing of ``ansible-core``

> features under active development. As described in the :ref:`roadmap <roadmaps>` for a given release, development is

Reviewer: ```suggestion
features under active development. As described in the `roadmap <https://docs.ansible.com/ansible-core/devel/roadmap/ansible_core_roadmap_index.html>`_  for a given release, development is
```

Reviewer: Another oddity - the old roadmaps is no longer used. Core and Ansible have different roadmaps. So pointing to the core roadmap directly here. I still need to figure out why a ref to the core roadmap doesn't work in the Ansible 5 etc package docs.

typically split into three phases of decreasing duration, with larger and more invasive changes targeted to be merged to
``devel`` in earlier phases. The ``milestone`` branch is updated to the contents of ``devel`` at the end of each
development phase. This allows testing of semi-stable unreleased features on a predictable schedule without the exposure

> to the potential instability of the daily commit "firehose" from ``devel``. When a release reaches the Beta 1 milestone,

Reviewer: ```suggestion
to the potential instability of the daily commit "fire hose" from ``devel``. When a release reaches the Beta 1 milestone,
```

the ``milestone`` branch will be updated to the first ``devel`` commit after the version number has been increased.
Further testing of the same release should be done from the new ``stable-X.Y`` branch that was created. If a severe issue
that significantly affects community testing or stability is discovered in the ``milestone`` branch, the branch contents
may require unscheduled adjustment, but not in a way that prevents fast-forward updates (e.g., ``milestone``-only
commits will not be created or cherry-picked from ``devel``).

> For example, the ``milestone`` branch in 2.13 ``ansible-core`` roadmap is scheduled to update as follows (based on

Reviewer 4: Will new `roadmap` files include dates for development phases and a list of what will be included in each phase?
(as well as a backlink to this file)?

Reviewer 3: The roadmap already includes this in my PR, however the dates haven't been approved as of yet: https://github.com/ansible/ansible/pull/75843/files

> For example, the ``milestone`` branch in 2.13 ``ansible-core`` roadmap is scheduled to update as follows (based on

Reviewer: ```suggestion
The following example is for illustrative purposes only. Please check the most recent roadmap for accurate dates. For example, the ``milestone`` branch in 2.13 ``ansible-core`` roadmap is scheduled to update as follows:
```

> dates from the published 2.13 development roadmap):

Reviewer: ```suggestion
```

Reviewer: Suggesting these changes so we don't have to maintain roadmap dates in 2 places for 2.13...

> dates from the published 2.13 development roadmap):

Reviewer 4: Should these dates move to roadmap?

> * 18-Oct-2021: 2.13 Development Phase 1 begins; ``milestone`` contents are updated to 2.12.0b1 with version number set to

Reviewer 3: I tweaked this date slightly in the merged PR.  Beta 1 happened Sept 27, but I didn't have Dev Phase 1 and Beta 1 linked.

```suggestion
* 27-Sep-2021: 2.13 Development Phase 1 begins; ``milestone`` contents are updated to 2.12.0b1 with version number set to
```
